### PR TITLE
Add new deploy app command

### PR DESCRIPTION
**Pull request by nenb**
This PR adds a new command: `deploy-app`.

`deploy-app`

- Opens the URL for the `create-app` page in a new browser tab. Also passes a query parameter for `filepath` that is generated from the current workspace in JLab. Accepts an argument `origin` ('menu', 'context-menu' etc) that allows controlling what icon is displayed.

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

## Any other comments?

Original issue: https://github.com/nebari-dev/jhub-apps/issues/472

### CODE REVIEW

File: junit.xml
[the remark this answers is not in the document]
Comment (nenb) [reply junit.xml:1]: Removed.

File: style/icons/deploy-app.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
...
...6C4.53026 4.15976 5.21026 3.94976 5.87026 4.07976L7.19026 4.34976ZM8.68026 14.5098C8.98026 14.8098 9.42026 14.8898 9.80026 14.7098C10.9603 14.1698 13.4503 12.8998 15.0603 11.2898C19.6503 6.69976 19.6903 2.95976 19.4203 1.35976C19.3503 0.959761 19.0303 0.639761 18.6303 0.569761C17.0303 0.299761 13.2903 0.339761 8.70026 4.92976C7.09026 6.53976 5.83026 9.02976 5.28026 10.1898C5.10026 10.5698 5.19026 11.0198 5.48026 11.3098L8.68026 14.5098ZM15.6503 12.8098C13.3603 14.8498 10.0703 16.2498 9.76026 16.3798L10.7303 18.6398C11.0103 19.2898 11.8603 19.4498 12.3603 18.9498L15.3703 15.9398C15.8403 15.4698 16.0503 14.7898 15.9203 14.1298L15.6503 12.8098ZM6.94026 15.4098C7.14026 16.4698 6.79026 17.4498 6.12026 18.1198C5.35026 18.8898 2.96026 19.4598 1.41026 19.7598C0.720261 19.8898 0.11026 19.2798 0.24026 18.5898C0.54026 17.0398 1.10026 14.6498 1.88026 13.8798C2.55026 13.2098 3.53026 12.8598 4.59026 13.0598C5.76026 13.2798 6.72026 14.2398 6.94026 15.4098ZM11.0003 6.99976C11.0003 5.89976 11.9003 4.99976 13.0003 4.99976C14.1003 4.99976 15.0003 5.89976 15.0003 6.99976C15.0003 8.09976 14.1003 8.99976 13.0003 8.99976C11.9003 8.99976 11.0003 8.09976 11.0003 6.99976Z" fill="black" fill-opacity="0.54"/>
```
[the remark this answers is not in the document]
Comment (nenb) [reply style/icons/deploy-app.svg]: Is this what you had in mind: https://github.com/nebari-dev/jupyterlab-jhub-apps/blob/7a3636d4fce76a8a612039be5b85b0304be7e045/style/icons/deploy-app.svg?short_path=ad22341

File: style/icons/deploy-app.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg" class="jp-icon3">
```
Comment (krassowski) [inline style/icons/deploy-app.svg:1]:
```suggestion
<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```

The class only needs to be set on `<path>`, not on the `<svg>`

File: style/icons/deploy-app.svg (hunk 1 above)
[the remark this answers is not in the document]
Comment (krassowski) [reply style/icons/deploy-app.svg]: Yeah, mostly. I left an inline comment.

**Comment by krassowski** [comment]
Updating snapshots failed because of timeouts I think:

https://github.com/nebari-dev/jupyterlab-jhub-apps/actions/runs/11060426290/job/30730934981

Screenshot from 2024-09-27 10-20-38

File: ui-tests/tests/jupyterlab_jhub_apps.spec.ts-snapshots/notebook-toolbar-before-click-linux.png
[the remark this answers is not in the document]
Comment (nenb) [reply ui-tests/tests/jupyterlab_jhub_apps.spec.ts-snapshots/notebook-toolbar-before-click-linux.png:1]: I will add the "Deploy App" label in the toolbar in a follow-up PR, just to try and get this PR merged.

**Comment by nenb** [comment]
bot please update snapshots

**Comment by nenb** [comment]
bot please update snapshots

**Review by krassowski — APPROVED** [review]
Thank you @nenb!

> For context, this PR was previously reviewed in https://github.com/nebari-dev/jupyterlab-nebari-mode/pull/7.

**Pull request merged**